flocker/provision/_install.py

```diff
@@ -40,9 +40,118 @@
                 "centos/clusterhq-release$(rpm -E %dist).noarch.rpm".format(
                     archive_bucket=ARCHIVE_BUCKET,
                     ),
+    # FLOC-1828 TODO - use ARCHIVE_BUCKET rather than clusterhq-archive-testing
+    # This needs a release to be done to add .deb files to clusterhq-archive
+    'ubuntu-14.04': 'https://clusterhq-archive-testing.s3.amazonaws.com/'
+                'ubuntu/14.04/$(ARCH)'.format(archive_bucket=ARCHIVE_BUCKET),
 }
 
 
+def task_client_installation_test():
+    """
+    Check that the CLI is working.
+    """
+    return run_from_args(['flocker-deploy', '--version'])
+
+
+def task_install_cli(distribution, package_source=PackageSource()):
+    """
+    Install flocker CLI on a distribution.
+
+    The ClusterHQ repo is added for downloading latest releases.  If
+    ``package_source`` contains a branch, then a BuildBot repo will also
+    be added to the package search path, to use in-development packages.
+    Note, the ClusterHQ repo is always enabled, to provide dependencies.
+
+    :param bytes distribution: The distribution the node is running.
+    :param PackageSource package_source: The source from which to install the
+        package.
+    """
+    if package_source.branch:
+        # A development branch has been selected - add its Buildbot repo
+        use_development_branch = True
+        result_path = posixpath.join(
+            '/results/omnibus/', package_source.branch, distribution)
+        base_url = urljoin(package_source.build_server, result_path)
+    else:
+        use_development_branch = False
+
+    if distribution == 'ubuntu-14.04':
```

Comment:
This if/else is way too big. Split it up into two functions, one for apt, one for rpm.